pennylane/measurements.py

```diff
@@ -764,7 +794,13 @@ def circuit(x):
             )
         wires = qml.wires.Wires(wires)
 
-    return MeasurementProcess(Counts, obs=op, wires=wires)
+    meas_process = MeasurementProcess(Counts, obs=op, wires=wires)
+    if all_outcomes:
+        meas_process.return_type.all_outcomes = True
+    else:
+        meas_process.return_type.all_outcomes = False
```

Comment:
These setters are not very pythonic... we should change this if we can.

I can't think of a different implementation besides adding a new `ObservableReturnType`. @antalszava do you think it would be excessive to add a new enum type? If not we can keep this as it is.

Reply:
I think so, but we can definitely do that if we think it'll improve code quality. No issues with having e.g., `Counts` and `AllCounts`.

Reply:
@AlbertMitjans, sure, I'll switch it to AllCounts and go through the other comments tomorrow.